solr/core/src/java/org/apache/solr/core/HttpSolrClientProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.core;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.solr.client.solrj.impl.Http2SolrClient;
+import org.apache.solr.common.util.IOUtils;
+import org.apache.solr.metrics.SolrMetricManager;
+import org.apache.solr.metrics.SolrMetricsContext;
+import org.apache.solr.security.HttpClientBuilderPlugin;
+import org.apache.solr.update.UpdateShardHandlerConfig;
+import org.apache.solr.util.stats.InstrumentedHttpListenerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provider of the default SolrClient implementation.
+ *
+ * @lucene.internal
+ */
+class HttpSolrClientProvider {
```

Comment:
We were looking for a new home for this client and didn't want to add the burden of creating and managing it to CoreContainer. So, we created a new class for that purpose. However, since we're still in the initial phase and experimenting with it, we don't want to expose this class publicly. That's why it's package-private and marked with @lucene.internal. In fact, I'm considering making this class final as well.

Reply:
It takes `UpdateShardHandlerConfig` for setting such as IdleTimeout, connectionTimeout and maxConnectionPerDestination. However, I do think that this client should have its own configuration to configure those values. Because right now all of three Http2SolrClient -- update, recovery and default -- are sharing same configuration.